### ISSUE TITLE
aiven_vpc_peering_connection datasource doesn't seem to be working with provider 2.5.0

### ISSUE THREAD

**Issue by Hi-Fi** [aiven/terraform-provider-aiven#670]
Just updated provider (from 2.1.11), and trying to get plan with Terraform caused error:
```
Error: Value for unconfigurable attribute

  on main.tf line 389, in data "aiven_vpc_peering_connection" "aiven_vpc_peering_azure":
 389:   peer_resource_group = var.az_resource_group

Can't configure a value for "peer_resource_group": its value will be decided
automatically based on the result of applying this configuration.
```

When tried to remove that value, error changed to form:
```
Error: cannot get an Azure VPC peering connection without `peer_resource_group`

  on main.tf line 385, in data "aiven_vpc_peering_connection" "aiven_vpc_peering_azure":
 385: data "aiven_vpc_peering_connection" "aiven_vpc_peering_azure" {
```

Is there some bug, or wrong usage?

**Comment by Hi-Fi** [comment]
This same happens also with 2.4.0, which is the oldest version that can be used due to migration from Elasticsearch to OpenSearch.

**Comment by Hi-Fi** [comment]
Current workaround is to copy needed value (`state_info.to-network-id`) directly to Terraform, but it's a bit dirty.

**Comment by mhoffm-aiven** [comment]
Seems to be a bug in the datasource I think. The peer resource group is a required to fetch the resource but cannot be set in the datasource.

**Comment by ivan-savciuc** [comment]
This issue is fixed in `v3.2.1`

**Comment by ivan-savciuc** [comment]
As a side note, Aiven no longer supports Elasticsearch.
